### ISSUE TITLE
Possible inconsistency with NetworkInterfaceAvailable waiter state

### ISSUE THREAD

**Issue by aBurmeseDev** [aws/aws-sdk#301]
Original Issue: [boto/botocore/issues/2717](https://github.com/boto/botocore/issues/2717)

**Comment by aBurmeseDev** [comment]
P67535117

**Comment by aBurmeseDev** [comment]
Hi @romos - we just heard back from a member of service team that `NetworkInterfaceAvailable` is for async operations like `DetachNetworkInterface`. This waiter waits on the ENI status to change from in-use to available. Thus, the `InvalidNetworkInterfaceID.NotFound` is configured correctly as a failure case.

`CreateNetworkInterface` is a synchronous operation and the full ENI description is returned in the API response. There is no need to use a waiter for this operation.

Also refer to this [docs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/query-api-troubleshooting.html#eventual-consistency) for why InvalidNetworkInterfaceID.NotFound could happen in this case. 

Please let us know if you have any further questions, thanks!

**Comment by pspacek** [comment]
Hello,

I'm not the original reported, but I'm facing the same issue couple times a week, including today.

I understand that `CreateNetworkInterface` is supposed to be synchronous, but _sometimes_ it does not work like that. The call returns ENI description just fine, but attempting `EC2.NetworkInterface.attach()` operations with the new ENI ID sometimes fails.

If the `CreateNetworkInterface` call was reliably synchronous `EC2_CLIENT.create_network_interface()` call followed by `EC2_CLIENT.get_waiter("network_interface_available").wait()` with the new ENI ID should not ever fail with `InvalidNetworkInterfaceID.NotFound` - but it keeps happening.

This can be interpreted this in two ways:
- Either the waiter has incorrect assumption and should handle NotFound as well
- Or the `CreateNetworkInterface` call is not working synchronously 100% of the time

I'm happy to provide more information about individual failures. Is there set of identifiers which would help you to narrow down what's happening?

**Comment by aBurmeseDev** [comment]
Hello @pspacek - sorry to hear you're seeing similar behavior. 

Since it's related to service API rather than SDK which we support here, would you please reach out directly to service team via [AWS Support](https://aws.amazon.com/contact-us/) or open a new issue on this repo and we can contact them on your behalf?

Thanks

**Comment by pspacek** [comment]
Hello. It's unclear to me what should I report. Are you saying that the waiter is 100% correct in expecting the call to be synchronous but the endpoint misbehaves and is not actually synchronous?
